eventio/base.py

```diff
@@ -51,8 +53,14 @@ def __iter__(self):
 
     def __next__(self):
         self._filehandle.seek(self._next_header_pos)
-        header = read_next_header_toplevel(self)
-        self._next_header_pos = self._filehandle.tell() + header.length
+        read_sync_marker(self)
+        header = read_header(
+            self,
+            toplevel=True,
```

Comment:
oncle bob says something about boolean parameters to functions .... according to him, such functions break the single responsibility principle, since they quite obviously do two things.

If I remember correctly I tried to get rid of this boolean parameter once and replace the one function with a boolean parameter with two functions...

Maybe that was not good?

Reply:
This Switch simply adds 4 bytes more to header size. And I tend to disagree with the small functions craze. I really disliked the names before, because header top level called header sub level, that makes no sense

Reply:
Yep ... I see the point.